terraform/api_gateway.tf.json.template.py

```diff
@@ -17,17 +28,17 @@ class Lambda:
     Represents a AWS Lambda function fronted by an AWS API Gateway.
     """
     name: str  # the name of the Lambda, e.g. 'service'
-    gateway_id: str  # the ID of the API Gateway fronting the service
     domains: List[str]  # a list of public domain names that the Lambda should be exposed at
+    policy: str  # AWS Policy for the lambda function
 
     @classmethod
     def for_name(cls, name):
         return cls(name=name,
-                   gateway_id=aws.api_gateway_id(config.qualified_resource_name(name)),
                    domains=[
                        config.api_lambda_domain(name),
                        *config.api_lambda_domain_aliases(name)
-                   ])
+                   ],
+                   policy=json.dumps(policies[name]))
```

Comment:
Consider replacing this, the dict up top and the static import with a dynamic import here, but only if it reduces the amount of code.

Reply:
It works and I've done it.